docs/usage-examples.md

```diff
@@ -0,0 +1,84 @@
+# Providing usage examples of orbs
+_The `examples` stanza is available in configuration version 2.1 and later_
+
+As an author of an orb, you may want to document examples of using it in a CircleCI config file, not only to provide a starting point for new users, but also to demonstrate more complicated use cases.
+
+## Simple examples
+Given the following orb:
+
+```yaml
+version: 2.1
+description: A foo orb
+
+commands:
+  hello:
+    description: Greet the user politely
+    parameters:
+      username:
+        type: string
+        description: A name of the user to greet
+    steps:
+      - run: "echo Hello << parameters.username >>"
+```
+
+...you can supply an additional `examples` stanza like this:
+
+```yaml
+examples:
+  simple_greeting:
+    description: Greeting a user named Anna
+    usage:
+      version: 2.1
+      orbs:
+        foo: bar/foo@1.2.3
+      jobs:
+        build:
+          machine: true
+          steps:
+            - foo/hello:
+                username: "Anna"
+```
+
+Please note that `examples` can contain multiple keys at the same level as `simple_greeting`, allowing for multiple examples.
+
+## Expected usage results
+
+The above usage example can be supplemented with a `result` key, demonstrating what the configuration will look like after expanding the orb with its parameters:
+
+```yaml
+examples:
+  simple_greeting:
+    description: Greeting a user named Anna
+    usage:
+      version: 2.1
+      orbs:
+        foo: bar/foo@1.2.3
+      jobs:
+        build:
+          machine: true
+          steps:
+            - foo/hello:
+                username: "Anna"
+    result:
+      version: 2.1
+      jobs:
+        build:
+          machine: true
+          steps:
+          - run:
+              command: echo Hello Anna
+      workflows:
+        version: 2
+        workflow:
+          jobs:
+          - build
+```
+
+When validating orb's source, the `result` map will be automatically compared with the actual output of compiling `usage`, and an error will be raised in case of a mismatch.
```

Comment:
Nit: The "result" comparison is possibly too large in scope for our initial work. We should rephrase this to say "in the future we may add automated example compilation and compare the output with `result`".

Reply:
I would go even further and remove `result` from the schema completely, initially - if we don't, we risk "breaking" already published orbs in the future when we roll this check out.

@ndintenfass, do you have objections to that?

Reply:
As `result` support is already deployed and it's trivial to get actual compilation output with the CLI, I'm only amending the docs to warn about a possibility of automated checks in the future.

Reply:
That sounds like a good middle ground. Thanks for updating this to match the merged work!